src/dataset/custom_dataset.py

```diff
@@ -579,7 +584,7 @@ def callback(exps, logs, seeds):
         return self
 
     def __len__(self):
-        return self.num_shards * self.eps_per_shard
+        return min(self.num_shards * self.eps_per_shard, self.args["num_samples"])
```

Comment:
this is just a suggestion - decided for this as it worked with the full length episodes